cmd/antrea-agent/options_test.go

```diff
@@ -242,9 +266,11 @@ func TestOptionsValidateMulticastConfig(t *testing.T) {
 				Multicast: agentconfig.MulticastConfig{
 					IGMPQueryVersions: tt.igmpQueryVersions},
 			}}
-			err := o.validateMulticastConfig()
+			err := o.validateMulticastConfig(tt.encryptionMode)
 			assert.Equal(t, tt.expectedErr, err)
-			assert.Equal(t, tt.expectedVersions, o.igmpQueryVersions)
+			if err != nil {
```

Comment:
instead of this, we usually use `require` to verify the error. This way the test stops early if the `err` value is not what we expect, and we "skip" subsequent assertions.

Reply:
In the test, we have designed both happy path and bad paths. This check is valid only with happy path (err is nil ). If we use `assert.Require`, it may only work with the happy cases.

Reply:
sorry I think I wasn't clear, I was just suggesting replacing `assert.Equal(t, tt.expectedErr, err)` with `require.Equal(t, tt.expectedErr, err)`